gammapy/utils/metadata.py

```diff
@@ -193,9 +193,10 @@ class ObsInfoMetaData(MetaData):
         The observation mode.
     """
 
+    model_config = ConfigDict(coerce_numbers_to_str=True)
     _tag: ClassVar[Literal["obs_info"]] = "obs_info"
 
-    obs_id: int
+    obs_id: str
```

Comment:
Why is it changed to `str`?

Reply:
We discussed previously that obs_ids don't need to be necessarily integers, and can be strings, and thus the MapDatasetMetaData was taking strings for obs_id

Having a `str` accepts both `int` and `str` on input

Reply:
Currently the gadf requirement is that this is an int. That's why I kept this constrain when introducing the `ObsInfoMetaData`. 
Unless there is a strong constraint for the `MapDatasetMetaData` I'd rather keep the same solution and update later if necessary. No?

Reply:
I have no opinions on this.. I can then change the behaviour for the MapDataset as well...

Reply:
@bkhelifi The current implementation allows to read `int` as seen in the tests

Reply:
Indeed, but pydantic is changing the type to `str`. 
For now, I think it is simpler to stick to `int`.